Sources_v3/StreamChatUI/ChatChannel/ChatChannelMessageComposerView.swift

```diff
@@ -21,152 +24,156 @@ open class ChatChannelMessageComposerView<ExtraData: UIExtraDataTypes>: UIInputV
+        ).withoutAutoresizingMaskConstraints
+    }()
+    
+    public private(set) lazy var shrinkInputButton: UIButton = {
+        ChatSquareButton(
+            image: UIImage(named: "shrinkInputArrow", in: .streamChatUI)!
+        ).withoutAutoresizingMaskConstraints
+    }()
+    
+    public private(set) lazy var stateIcon: UIImageView = {
+        let imageView = UIImageView().withoutAutoresizingMaskConstraints
+        imageView.contentMode = .scaleAspectFit
+        imageView.tintColor = .squareButtonTint
+        imageView.widthAnchor.constraint(equalTo: imageView.heightAnchor, multiplier: 1).isActive = true
+        return imageView
+    }()
+    
+    public private(set) lazy var dismissButton: UIButton = {
+        ChatSquareButton(
+            image: UIImage(named: "dismissInCircle", in: .streamChatUI)!,
+            tintColor: .squareButtonTint
+        ).withoutAutoresizingMaskConstraints
+    }()
+    
+    public private(set) lazy var titleLabel: UILabel = {
+        let label = UILabel().withoutAutoresizingMaskConstraints
+        label.textAlignment = .center
+        label.font = .preferredFont(forTextStyle: .headline)
+        label.adjustsFontForContentSizeCategory = true
+        return label
     }()
     
     // MARK: - Init
     
-    public required init(
-        uiConfig: UIConfig<ExtraData> = .default
-    ) {
-        self.uiConfig = uiConfig
-        
+    public required init() {
         super.init(frame: .zero, inputViewStyle: .default)
-        
-        commonInit()
     }
     
     public required init?(coder: NSCoder) {
-        uiConfig = .default
-        
         super.init(coder: coder)
-        
-        commonInit()
     }
     
-    public func commonInit() {
-        embed(container)
-
-        setupAppearance()
-        setupLayout()
+    // MARK: - Overrides
+    
+    override open func didMoveToSuperview() {
+        applyDefaultAppearance()
+        setUpAppearance()
+        setUpLayout()
+    }
+    
+    override open var intrinsicContentSize: CGSize {
+        let size = CGSize(
+            width: UIView.noIntrinsicMetric,
+            height: container.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize).height
+        )
+        return size
+    }
+    
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+        
+        container.centerStackView.clipsToBounds = true
+        container.centerStackView.layer.cornerRadius = 25
+        container.centerStackView.layer.borderWidth = 1
+        container.centerStackView.layer.borderColor = UIColor.systemGray.withAlphaComponent(0.3).cgColor
+        
+        layer.shadowColor = UIColor.systemGray.cgColor
+        layer.shadowOpacity = 1
+        layer.shadowOffset = .zero
+        layer.shadowRadius = 0.5
     }
     
     // MARK: - Public
     
-    open func setupAppearance() {
+    open func setUp() {}
+    
+    open func defaultAppearance() {
+        attachmentsViewHeight = 80
```

Comment:
maybe this belongs to`setupLayout`? Maybe I didn't understand our `UIConfig & Structur` guide...

Reply:
Same as https://github.com/GetStream/stream-chat-swift/pull/629#discussion_r537820462